### ISSUE TITLE
Add rule for slf4j-2's new fluent API

### ISSUE THREAD

**Issue by mattnelson** [KengoTODA/findbugs-slf4j#470]
slf4j has been released with a new fluent API.
https://www.slf4j.org/manual.html#fluent

Suggestion is to add a new rule enforcing the below expectation.

> When using the fluent API, you must terminate the invocation chain by calling one of the log() method variants. Forgetting to call any of the log() method variants will result in no logging regardless of the logging level. Fortunately, many IDEs will alert you with a "No return value" compiler warning.

Confirmed that IntelliJ gives some notice of the issue
<img width="432" alt="image" src="https://user-images.githubusercontent.com/1894657/188994047-5e21ff7e-a642-414a-80ab-77bc58821452.png">

But I do not see any javac output indicating an issue.

**Comment by mattnelson** [comment]
Linking similar request for errorprone-slf4j

https://github.com/KengoTODA/errorprone-slf4j/issues/143

**Comment by KengoTODA** [comment]
here is a PR to realize this feature, please consider to vote on it:

https://github.com/qos-ch/slf4j/pull/279

**Comment by mattnelson** [comment]
Looks like support was added in https://github.com/qos-ch/slf4j/commit/4e4e56a2df730cdb6877449b6341279208f20a78

My original test was in a test, which I forgot was not setup for spotbugs yet. When I added this bug to code that is scanned by spotbugs it does produce
https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#rv-method-ignores-return-value-rv-return-value-ignored

```
[ERROR] Medium: Return value of org.slf4j.spi.LoggingEventBuilder.setCause(Throwable) ignored in class.method() [class] At class.java:[line 87] RV_RETURN_VALUE_IGNORED
```

Gonna close this out as spotbugs and error prone have equivalent checkers on CheckReturnValue.
https://errorprone.info/bugpattern/CheckReturnValue

I wasn't able to get errorprone to detect the bug, but that discussions doesn't belong here.
